lib/http2/hpack.c

```diff
@@ -1067,6 +1067,7 @@ void h2o_hpack_flatten_response(h2o_buffer_t **buf, h2o_hpack_header_table_t *he
 
     /* setup the frame headers */
     fixup_frame_headers(buf, start_at, H2O_HTTP2_FRAME_TYPE_HEADERS, stream_id, max_frame_size, 0);
+    return (*buf)->size - start_at - H2O_HTTP2_FRAME_HEADER_SIZE;
```

Comment:
When the response headers are large, they might be split into a HEADERS frame followed by a series of CONTINUATION frames, in which case the overhead will be more than `H2O_HTTP2_FRAME_HEADER_SIZE`.

Therefore, maybe something like untested code below (i.e., calculate before calling `fix_frame_headers`, while we work under the assumption that there would be only one H2 frame):

```suggestion
    size_t headers_size = (*buf)->size - start_at - H2O_HTTP2_FRAME_HEADER_SIZE;

    /* setup the frame headers */
    fixup_frame_headers(buf, start_at, H2O_HTTP2_FRAME_TYPE_HEADERS, stream_id, max_frame_size, 0);

    return header_size;
```